zebra-network/src/peer_set/set.rs

```diff
@@ -152,6 +161,31 @@ where
         });
     }
 
+    fn check_for_background_errors(&mut self, cx: &mut Context) -> Result<(), BoxedStdError> {
+        if self.guards.is_empty() {
+            match self.handle_rx.try_recv() {
+                Ok(handles) => {
+                    for handle in handles {
+                        self.push_join_handle(handle);
+                    }
+                }
+                Err(tokio::sync::oneshot::error::TryRecvError::Closed) => unreachable!(
+                    "try_recv will never be called if the futures have already been received"
+                ),
+                Err(tokio::sync::oneshot::error::TryRecvError::Empty) => return Ok(()),
```

Comment:
tioli: a `use tokio::sync::oneshot::error::TryRecvError` above the match statement might make this slightly easier to read